upup/pkg/fi/cloudup/new_cluster.go

```diff
@@ -745,10 +747,32 @@ func setupZones(opt *NewClusterOptions, cluster *api.Cluster, allZones sets.Stri
 			}
 			cluster.Spec.Networking.Subnets = append(cluster.Spec.Networking.Subnets, *subnet)
 		}
-		zoneToSubnetMap[zoneName] = subnet
+		zoneToSubnetsMap[zoneName] = append(zoneToSubnetsMap[zoneName], subnet)
 	}
 
-	return zoneToSubnetMap, nil
+	if len(opt.NetworkCIDRs) > 1 {
+		for i, cidr := range opt.NetworkCIDRs[1:] {
+			zoneName := opt.Zones[i%len(opt.Zones)]
+			subnetName := fmt.Sprintf("%s-%d", zoneName, i+1)
```

Comment:
I think this is pretty specific to AWS.  Also the way we round-robin around the zones is convenient, but might be surprising (along with the numbering).

Should we feature-flag this feature (and/or restrict to AWS only at first)?

Reply:
One possibility would be to put the `-%d` suffix on the first subnet as well when `len(opt.NetworkCIDRs) > 1` but on AWS the primary CIDR of a VPC does have slightly different semantics.

Reply:
I agree about the slightly different semantics. This is why I preferred to have different names for additional network CIDRs.